### ISSUE TITLE
Bug: fail to test & upload attachment to S3

### ISSUE THREAD

**Issue by wingkwong** [nocodb/nocodb#307]
**Steps To Reproduce**

**1. Fail to test** 
1. Go to App Store and configure S3
2. Click Test
3. It shows "Successfully tested plugin settings"
4. Go to S3 bucket. No test images are uploaded.

**2. Fail to upload attachment**
1. Create a table with a column with an attachment type
2. Upload an image 
3. It shows "Cannot read property 'data' of undefined"

**Expected behavior**
1. Test images should be uploaded to target S3. If there is a connection issue, users should know it from the return message. 
2. Attachments should be able to uploaded to target S3. 

Node: **v12.22.1**
Arch: **x64**
Platform: **linux**
Docker: **true**
Database: **pg**
ProjectOnRootDB: **false**
RootDB: **pg**
PackageVersion: **0.9.37**

**Comment by wingkwong** [comment]
when I click "Test", behind the scene it fires a request to backend with operation xcPluginTest. I tracked it down to NcMetaMgr and found that the logic for Storage is actually commented out since the first version. See [here](https://github.com/nocodb/nocodb/blob/master/packages/nocodb/src/lib/noco/meta/NcMetaMgr.ts#L2857).

Not sure if the logic is here or have been moved to somewhere.

**Comment by wingkwong** [comment]
Thanks a lot. Verified that I could upload images successfully if I have the following settings. 

![image](https://user-images.githubusercontent.com/35857179/123813458-7ed87900-d927-11eb-8331-4d16af65a9c7.png)

Currently, the ACL has to be ``public-read``.  I think it would be better to have other options in UI so that we could upload images while keeping the objects in S3 private.

Besides, in case someone encounters ``Error SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your key and signing method.``, it is most likely your Secret Access Key contains characters like ``/``.
